src/main/java/org/tridiots/ipc/Client.java

```diff
@@ -1,41 +1,37 @@
 package org.tridiots.ipc;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.io.IOException;
-import java.io.ObjectInputStream;
-import java.io.ObjectOutputStream;
-import java.io.Serializable;
 import java.net.InetSocketAddress;
-import java.net.Socket;
+import java.nio.channels.SocketChannel;
 
 public class Client {
-
-    private Socket socket;
-    private ObjectInputStream ois;
-    private ObjectOutputStream oos;
+    private static final Logger logger = LoggerFactory.getLogger(Client.class);
+    private InetSocketAddress addr;
+    private SocketChannel sendChannel;
 
     public Client(InetSocketAddress addr) {
-        try {
-            this.socket = new Socket(addr.getAddress(), addr.getPort());
-            // caution: we must create outputsream before inputstream.
-            // see: https://stackoverflow.com/questions/5658089/java-creating-a-new-objectinputstream-blocks
-            this.oos = new ObjectOutputStream(this.socket.getOutputStream());
-            this.ois = new ObjectInputStream(this.socket.getInputStream());
-        } catch (IOException e) {
-            System.out.println("connection failed");
-            e.printStackTrace();
-        }
+        this.addr = addr;
     }
 
     public Object call(String methodName, Class<?>[] paramTypes, Object[] params) {
         Param param = new Param(methodName, paramTypes, params);
+
         try {
-            oos.writeObject(param);
-            return ois.readObject();
+            this.sendChannel = SocketChannel.open(addr);
+            //this.sendChannel.configureBlocking(false);
+            SocketObjectUtil.sendObject(sendChannel, param);
+            sendChannel.socket().shutdownOutput();
+            Object result = SocketObjectUtil.receiveObject(sendChannel);
```

Comment:
why use a `send` channel to receive ? might use a better name for `sendChannel`.